### PR TITLE
extensions observability

### PR DESCRIPTION
**Pull request by akosyakov**
Supporting for https://github.com/gitpod-io/gitpod/pull/12539. See this PR for explanation and how to test.

### CODE REVIEW

File: src/vs/gitpod/node/gitpodInsightsAppender.ts [hunk 1]
```diff
@@ -165,22 +91,176 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
...
+		return {
+			serverToken,
+			gitpodHost,
+			gitpodApiEndpoint,
+			workspaceId,
+			instanceId,
+		};
+	}
+
+	// TODO(ak) publihs to Segment directly to production/staging untrusted instead, use server api only to resolve a user
```
Comment (andreafalzetti) [inline src/vs/gitpod/node/gitpodInsightsAppender.ts:231]:
```suggestion
	// TODO(ak) publish to Segment directly to production/staging untrusted instead, use server api only to resolve a user
```

**Review by andreafalzetti — APPROVED** [review]
Tested on main PR 💯 

Code changes LGTM

File: src/vs/workbench/services/telemetry/browser/telemetryService.ts [hunk 2]
```diff
@@ -67,7 +67,8 @@ export class TelemetryService extends Disposable implements ITelemetryService {
 		remoteAgentService: IRemoteAgentService
 	) {
 		const telemetrySupported = supportsTelemetry(productService, environmentService) && productService.aiConfig?.ariaKey;
-		if (telemetrySupported && this.impl === NullTelemetryService && this.telemetryLevel.value !== TelemetryLevel.NONE) {
+
+		if (telemetrySupported/*  && this.impl === NullTelemetryService && this.telemetryLevel.value !== TelemetryLevel.NONE  see https://github.com/microsoft/vscode/pull/157279/files#r959575970*/) {
```
Comment (jeanp413) [inline src/vs/workbench/services/telemetry/browser/telemetryService.ts:71]: Rebase as this is already fixed on main branch

**Comment by jeanp413** [comment]
As I was on holiday on friday I didn't have time to do a proper review, will add comments so maybe they are addressed in the future

File: src/vs/gitpod/browser/gitpodInsightsAppender.ts [hunk 3]
```diff
@@ -6,14 +6,17 @@
...
-import { mapTelemetryData, SenderKind } from 'vs/gitpod/common/insightsHelper';
+import { mapMetrics, mapTelemetryData } from 'vs/gitpod/common/insightsHelper';
+import type { IDEMetric } from '@gitpod/ide-metrics-api-grpcweb';
+
+type SendMetrics = (metrics: IDEMetric[]) => Promise<void>;
 
 export class GitpodInsightsAppender implements ITelemetryAppender {
-	private _baseProperties: { appName: string; uiKind: 'web'; version: string };
+	private readonly _baseProperties: { appName: string; uiKind: 'web'; version: string };
+	private readonly devMode = this.productService.nameShort.endsWith(' Dev');
```
Comment (jeanp413) [inline src/vs/gitpod/browser/gitpodInsightsAppender.ts:16]: This doesn't look nice, should be moved to workbench.ts and if true then set `gitpodPreview` to `undefined` and use that as a check here

File: src/vs/gitpod/browser/gitpodInsightsAppender.ts [hunk 4]
```diff
@@ -22,15 +25,87 @@ export class GitpodInsightsAppender implements ITelemetryAppender {
...
+			console.error('failed to send IDE analytic:', e);
 		}
-		trackMessage.properties = {
-			...trackMessage.properties,
-			...this._baseProperties,
-		};
-		window.postMessage({ type: 'vscode_telemetry', event: trackMessage.event, properties: trackMessage.properties }, '*');
+	}
+
+	private async sendMetrics(eventName: string, data: any): Promise<void> {
```
Comment (jeanp413) [inline src/vs/gitpod/browser/gitpodInsightsAppender.ts:58]: The browser gitpodInsightsAppender.ts is just and edge for the reconnection events when we lose connection to the remote server, all other events will be send to the remote so not sure if we need the `sendMetrics` logic here

File: src/vs/gitpod/node/gitpodInsightsAppender.ts [hunk 5]
```diff
@@ -45,94 +48,15 @@ type GitpodConnection = Omit<GitpodServiceImpl<GitpodClient, GitpodServer>, 'ser
...
-	return gitpodService;
-}
-
 export class GitpodInsightsAppender implements ITelemetryAppender {
 
 	private _asyncAIClient: Promise<GitpodConnection> | null;
 	private _defaultData: { [key: string]: any } = Object.create(null);
 	private _baseProperties: { appName: string; uiKind: 'web'; version: string };
+	private readonly supervisor = new SupervisorConnection();
+	private readonly devMode = this.productName.endsWith(' Dev');
```
Comment (jeanp413) [inline src/vs/gitpod/node/gitpodInsightsAppender.ts:57]: Same as before, move to workbench.ts

**Pull request merged**